src/Mvc/src/Microsoft.AspNetCore.Mvc.NewtonsoftJson/ProblemDetailsConverter.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.Mvc.Formatters
+{
+    /// <summary>
+    /// A RFC 7807 compliant <see cref="JsonConverter"/> for <see cref="ProblemDetails"/>.
+    /// </summary>
+    public sealed class ProblemDetailsConverter : JsonConverter
```

Comment:
@JamesNK is this the best way to do this?

Reply:
I've never seen this approach before of having an internal class with attributes. It is a little inefficient to call Serialize/Deserialize inside the converter but in the context of a request the overhead is small, and this isn't a hot path. And the code is much easier to read compared to doing it manually with JsonReader/JsonWriter.

It looks fine to me.